src/modules/back-to-top/back-to-top.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Component, ElementRef, HostListener, Input, ViewChild } from '@angular/core';
+
+import { InputConverter } from '../shared';
+
+@Component({
+  selector: 'stache-back-to-top',
+  templateUrl: './back-to-top.component.html',
+  styleUrls: ['./back-to-top.component.scss']
+})
+export class StacheBackToTopComponent {
+  @Input()
+  @InputConverter()
+  public offset: number;
+
+  @ViewChild('backToTop')
+  public elementRef: ElementRef;
+
+  @HostListener('window:scroll')
+  public onWindowScroll() {
+    if (window.pageYOffset > this.offset) {
+      this.elementRef.nativeElement.classList.remove('hidden');
```

Comment:
Alternative, instead of worrying about the `elementRef`:

Component HTML:
```
<button 
  class="stache-back-to-top"
  [ngClass]="{ 'hidden': isHidden }"
  title="Scroll to the top of the page.">
  <i class="fa fa-chevron-down"></i>
</button>
```

Component script:
```
@HostListener('window:scroll')
public onWindowScroll() {
  this.isHidden = (window.pageYOffset < this.offset);
}
```